### PR TITLE
[SYCL][ESIMD] Preserve undef initializer for globals in ESIMDLowerVecArg pass

### PR DESCRIPTION
**Pull request by DenisBakhvalov**

### CODE REVIEW

**Review by bader — APPROVED** [review]

File: llvm/test/SYCLLowerIR/esimd_global_undef.ll [hunk 1]
```diff
@@ -0,0 +1,19 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt < %s -ESIMDLowerVecArg -S | FileCheck %s
+
```
Comment (kbobrovs) [inline llvm/test/SYCLLowerIR/esimd_global_undef.ll:3]: please add a comment what this test checks

File: llvm/test/SYCLLowerIR/esimd_global_undef.ll [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown-sycldevice"
+
+%"class._ZTSN2cl4sycl5INTEL3gpu4simdIiLi2512EEE.cl::sycl::INTEL::gpu::simd" = type { <2512 x i32> }
```
Comment (kbobrovs) [inline llvm/test/SYCLLowerIR/esimd_global_undef.ll:7]: Nit (for the sake of readability)
```suggestion
%"cl::sycl::INTEL::gpu::simd" = type { <2512 x i32> }
```

Comment (DenisBakhvalov) [reply llvm/test/SYCLLowerIR/esimd_global_undef.ll]: done

File: llvm/test/SYCLLowerIR/esimd_global_undef.ll (hunk 1 above)
Comment (DenisBakhvalov) [reply llvm/test/SYCLLowerIR/esimd_global_undef.ll:3]: done

**Comment by DenisBakhvalov** [comment]
@kbobrovs , ping

**Review by bader — APPROVED** [review]

**Pull request merged**